lib/Driver/ToolChains/Hcc.cpp

```diff
@@ -137,74 +153,151 @@ void HCC::CXXAMPAssemble::ConstructJob(Compilation &C, const JobAction &JA,
   C.addCommand(llvm::make_unique<Command>(JA, *this, Exec, CmdArgs, Inputs));
 }
 
-std::string temporaryReplaceLongFormGFXIp(const Compilation& C, std::string l)
-{ // Precondition: l = "AMD:AMDGPU:\d:\d:\d"
-  // TODO: this should be removed once we have transitioned all users to using
-  //       the short form. It is purposefully inefficient.
-  const auto t = l;
+namespace
+{
+    std::string temporaryReplaceLongFormGFXIp(
+        const Compilation& C, std::string l)
+    { // Precondition: l = "AMD:AMDGPU:\d:\d:\d"
+        // TODO: this should be removed once we have transitioned all users to
+        //       the short form. It is purposefully inefficient.
+        const auto t = l;
+
+        l.replace(0u, 3u, {'g', 'f', 'x'});
+        l.erase(std::copy_if(
+            l.begin() + 3u, l.end(), l.begin() + 3u, isdigit), l.end());
+        C.getDriver().Diag(diag::warn_drv_deprecated_arg) << t << l;
+
+        return l;
+    }
+
+    std::pair<std::vector<std::string>, int> detect_gfxip(
+        const Compilation& C, const ToolChain& TC)
+    {
+        std::vector<std::string> agents;
+
+        const auto e = C.getSysRoot() + "/opt/rocm/bin/rocm_agent_enumerator";
```

Comment:
Hello Siu Chi. Thanks for the feedback - good point! What I did locally, for the agent_enumerator unit tests, was to pass this as a macro define i.e. via target_compile_definitions. I think this would work here as well, by adding the ROCm path to the clang target. I was thinking that we may want something more flexible such as allowing one to pass the path to ROCm as an argument to the compilation, but I don't think we expose that type of granularity elsewhere so it may be a bridge too far for now.